internal/c99/headers/windows_amd64/builtin.h

```diff
@@ -0,0 +1,137 @@
+//  Copyright 2017 The CCIR Authors. All rights reserved.
+//  Use of this source code is governed by a BSD-style
+//  license that can be found in the LICENSE file.
+
+// +build ignore
+
+#ifndef _BUILTIN_H_
+#define _BUILTIN_H_
+
+#include "predefined.h"
+
+typedef struct {
+	void *_[2];		// Go *[]interface{}
+} *__builtin_va_list;
+
+#define _MSC_VER 1900
+#define _MSVCRT_
+#define _MSC_EXTENSIONS
+#define __int8 char
+#define BYTE unsigned char
+#define __int16 short
+#define __int32 int
+#define __int64 long long
+// disable SSE and other instrisics we dont support
+#define _X86INTRIN_H_INCLUDED 1
+#undef __SSE__
+#undef __SSE2__
+#undef __MMX__
+// bypass bug that fails for: #define MICROSOFT_WINDOWS_WINBASE_H_DEFINE_INTERLOCKED_CPLUSPLUS_OVERLOADS (_WIN32_WINNT >= 0x0502 || !defined (_WINBASE_))
+#define MICROSOFT_WINDOWS_WINBASE_H_DEFINE_INTERLOCKED_CPLUSPLUS_OVERLOADS 1
+#define __unaligned
+#define __inline__ inline
+#define __forceinline inline __attribute__((__always_inline__,__gnu_inline__))
+
+// TODO: __builtin_fprintf(stderr, "%s:%i.%s STUB %s called!\n", __FILE__, __LINE__, __func__, #x), __builtin_abort(), 0)
+#define abort_stubbed(x) 0
+
+// maybe: `#define __CRT__NO_INLINE` to remove some unnecessary stuff
+// Implementing __readgsqword as macro also prevents an anonymous union access 
+// (which isn't supported in C99 actually WTF)
+#define __readfsdword(x) abort_stubbed("__readfsdword")
+#define __readgsqword(x) abort_stubbed("__readgsqword")
+
+// TODO: ? not relevant for SQLite since we have mutexes
+#define _mm_mfence()
+
+#define __FUNCTION__ __func__
+#define __attribute(x)
+#define __attribute__(x)
+#define __builtin_choose_expr(a, b, c) (a) ? (b) : (c)
+#define __builtin_expect(exp, c) (exp)
+#define __builtin_offsetof(st, m) ((__SIZE_TYPE__)(&((st *)0)->m))
+#define __builtin_prefetch(addr, ...) (void)(addr)
+#define __builtin_signbit(x) (sizeof(x) == sizeof(float) ? __signbitf(x) : sizeof (x) == sizeof(double) ? __signbit(x) : __signbitl(x))
+#define __builtin_types_compatible_p(type1, type2) __builtin_types_compatible__((type1){}, (type2){})
+#define __complex__ _Complex
+#define __const const
+#define __extension__
+#define __inline inline
+#define __restrict restrict
+#define __roundup(n, mod) ((n + mod - 1) & ~(mod - 1))
+#define __volatile volatile
+
+typedef void *__FILE_TYPE__;
```

Comment:
Note that the linux variant now uses whatever the system headers define `FILE` to be. I don't think it's a problem ATM, though.